indico/modules/categories/controllers/display.py

```diff
@@ -347,3 +348,158 @@ def _process(self):
+            'subcategories': subcategories,
+            'start_dt': self.start_dt,
+            'end_dt': self.end_dt - relativedelta(days=1),  # Display a close-ended interval
+            'previous_day_url': self._other_day_url(self.start_dt - relativedelta(days=1)),
+            'next_day_url': self._other_day_url(self.start_dt + relativedelta(days=1)),
+            'previous_month_url': self._other_day_url(self.start_dt - relativedelta(months=1)),
+            'next_month_url': self._other_day_url(self.start_dt + relativedelta(months=1)),
+            'previous_year_url': self._other_day_url(self.start_dt - relativedelta(years=1)),
+            'next_year_url': self._other_day_url(self.start_dt + relativedelta(years=1))
+        }
+
+        if self.period == 'day':
+            return WPCategory.render_template('display/overview/day.html', self.category, events=events, **params)
+        elif self.period == 'week':
+            days = self._get_week_days()
+            template = 'display/overview/week.html'
+            params['previous_week_url'] = self._other_day_url(self.start_dt - relativedelta(days=7))
+            params['next_week_url'] = self._other_day_url(self.start_dt + relativedelta(days=7))
+        elif self.period == 'month':
+            days = self._get_calendar_days()
+            template = 'display/overview/month.html'
+
+        events_by_day = []
+        for day in days:
+            events_by_day.append((day, self._pop_head_while(lambda x: x.start_dt.date() <= day.date(), events)))
+
+        # Check whether all weekends are empty
+        hide_weekend = (not any(map(itemgetter(1), events_by_day[5::7])) and
+                        not any(map(itemgetter(1), events_by_day[6::7])))
+        if hide_weekend:
+            events_by_day = filter(lambda x: x[0].weekday() not in (5, 6), events_by_day)
+
+        return WPCategory.render_template(template, self.category, events_by_day=events_by_day,
+                                          hide_weekend=hide_weekend, **params)
+
+    def _get_week_days(self):
+        # Return the days shown in the weekly overview
+        return self._get_days(self.start_dt, self.end_dt)
+
+    def _get_calendar_days(self):
+        # Return the days shown in the monthly overview
+        start_dt = self.start_dt - relativedelta(days=self.start_dt.weekday())
+        end_dt = self.end_dt + relativedelta(days=(7 - self.end_dt.weekday()) % 7)
+        return self._get_days(start_dt, end_dt)
+
+    @staticmethod
+    def _get_days(start_dt, end_dt):
+        # Return all days in the open-ended interval
+        current_dt = start_dt
+        while current_dt < end_dt:
+            yield current_dt
+            current_dt = current_dt + relativedelta(days=1)
+
+    @staticmethod
+    def _pop_head_while(predicate, l):
+        # Pop the head of the list while the predicate is true and return the popped elements
+        res = []
+        while len(l) and predicate(l[0]):
```

Comment:
`while l and predicate(l[0])`

also, don't use lowercase-L as a name (PEP8 explicitly discourages `I` and `l` because of the ambiguity in most fonts). you could call it `lst` or `list_` instead